server/src/main/java/com/defold/extender/services/CocoaPodsService.java

```diff
@@ -0,0 +1,625 @@
+package com.defold.extender.services;
+
+import com.defold.extender.ExtenderException;
+import com.defold.extender.ExtenderUtil;
+import com.defold.extender.ProcessExecutor;
+import com.defold.extender.TemplateExecutor;
+import com.defold.extender.metrics.MetricsWriter;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.springframework.stereotype.Service;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.lang.StringBuilder;
+
+
+@Service
+public class CocoaPodsService {
+
+    public class ResolvedPods {
+        public List<PodSpec> pods = new ArrayList<>();
+        public File podsDir;
+        public File frameworksDir;
+        public String platformVersion;
+
+        public Set<String> getAllPodLibs(String platform) {
+            Set<String> libs = new HashSet<>();
+            for (PodSpec pod : pods) {
+                libs.addAll(pod.libraries);
+                if (platform.contains("ios")) {
+                    libs.addAll(pod.ios_libraries);
+                }
+                else if (platform.contains("osx")) {
+                    libs.addAll(pod.osx_libraries);
+                }
```

Comment:
You mean to add `libraries` to `ios_libraries` and to `osx_libraries` and return one or the other instead of building a new merged list each time the function is called?